src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -490,3 +530,206 @@ def execute_hdi_script_action(cmd, client, resource_group_name, cluster_name,
+    _extract_and_validate_autoscale_configuration(cluster)
+    autoscale_configuration = None
+    if no_wait:
+        return sdk_no_wait(no_wait, client.update_auto_scale_configuration, resource_group_name, cluster_name,
+                           autoscale_configuration)
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def list_timezones(cmd, client):
+    from .util import AUTOSCALE_TIMEZONES
+    return AUTOSCALE_TIMEZONES
+
+
+def create_autoscale_condition(cmd, client, resource_group_name, cluster_name, days, time, workernode_count,
+                               no_wait=False):
+    from azure.mgmt.hdinsight.models import AutoscaleRecurrence, AutoscaleSchedule, AutoscaleTimeAndCapacity
+    cluster = client.get(resource_group_name, cluster_name)
+    autoscale_configuration = _extract_and_validate_autoscale_configuration(cluster)
+
+    # try to add schedule condition
+    condition = AutoscaleSchedule(
+        days=days,
+        time_and_capacity=AutoscaleTimeAndCapacity(
+            time=time,
+            min_instance_count=workernode_count,
+            max_instance_count=workernode_count)
+    )
+    if autoscale_configuration.recurrence:
+        autoscale_configuration.recurrence.schedule.append(condition)
+    elif not autoscale_configuration.recurrence:
+        autoscale_configuration.recurrence = AutoscaleRecurrence(
+            schedule=[condition]
+        )
+    if no_wait:
+        return sdk_no_wait(no_wait, client.update_auto_scale_configuration, resource_group_name, cluster_name,
+                           autoscale_configuration)
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def update_autoscale_condition(cmd, client, resource_group_name, cluster_name, index, days=None, time=None,
+                               workernode_count=None, no_wait=False):
+    cluster = client.get(resource_group_name, cluster_name)
+    autoscale_configuration = _extract_and_validate_autoscale_configuration(cluster)
+    _validate_schedule_configuration(autoscale_configuration)
+
+    if index >= len(autoscale_configuration.recurrence.schedule):
+        raise CLIError(
+            'This cluster only has {} condition(s). Please specify a correct index which starts with 0.'.format(
+                len(autoscale_configuration.recurrence.schedule)))
+
+    # try to update schedule condition
+    if days:
+        autoscale_configuration.recurrence.schedule[index].days = days
+    if time:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.time = time
+    if workernode_count:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.min_instance_count = workernode_count
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.max_instance_count = workernode_count
+
+    if no_wait:
```

Comment:
Do we need this `if`? If `no_wait==False`. `sdk_no_wait` should be same with `client.update_auto_scale_configuration`

Reply:
> Do we need this `if`? If `no_wait==False`. `sdk_no_wait` should be same with `client.update_auto_scale_configuration`

Removed unnecessary `if no_wiat`.